src/utils/message_entity.rs

```diff
@@ -0,0 +1,116 @@
+use std::string::FromUtf16Error;
+
+use teloxide_core::types::MessageEntity;
+
+#[derive(Debug)]
+pub enum InvokeEntityError {
+    TooShortText { offset: usize, length: usize },
+    FromUTF16(FromUtf16Error),
+}
+
+impl From<FromUtf16Error> for InvokeEntityError {
+    fn from(e: FromUtf16Error) -> Self {
+        Self::FromUTF16(e)
+    }
+}
+
+impl std::fmt::Display for InvokeEntityError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            InvokeEntityError::TooShortText { offset, length } => write!(
+                f,
+                "Text is too short for entity with offset {} and length {}",
+                offset, length
+            ),
+            InvokeEntityError::FromUTF16(e) => e.fmt(f),
+        }
+    }
+}
+
+impl std::error::Error for InvokeEntityError {}
+
+fn invoke_entity_from_utf16(
+    text: &[u16],
+    entity: &MessageEntity,
+) -> Result<String, InvokeEntityError> {
+    let start = entity.offset;
+    let end = entity.offset + entity.length;
+    if text.len() < end {
+        Err(InvokeEntityError::TooShortText { offset: entity.offset, length: entity.length })
+    } else {
+        String::from_utf16(&text[start..end]).map_err(Into::into)
+    }
+}
+pub fn invoke_entity(text: &str, entity: &MessageEntity) -> Result<String, InvokeEntityError> {
```

Comment:
Is this really needed?
I think that we should only provide the "in batch" function.

Reply:
in some case you may need only one entity by type, for example, last mention in message.

Reply:
Since it requires +- the same amount of work, why not just convert all entities?

Reply:
in that case we need to keep in mind index of entitiy, which we are interested in

Reply:
Why?

Instead of "find what entity you want" and then "convert it" you can "convert all" and then "find what converted entity you want.